### PR TITLE
move delete button near edit button

### PR DESCRIPTION
**Pull request by KevinArtus**
[OO-FEATURE] Remove delete button on edit modal view and deplace it #near edit button
[OO-BCBREAK] Deleted deleteButton._tpl.twig

### CODE REVIEW

File: BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          url: params.url
+          method: 'DELETE'
+          success: (response) ->
+            if redirectUrl != undefined
+              refreshMenu(redirectUrl)
+            else
+              redirectUrl = appRouter.generateUrl 'showDashboard'
+              refreshMenu(redirectUrl)
+            return
+      )
```
Comment (itkg-nanne) [inline BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee]: line return at EOF

File: BackofficeBundle/Resources/public/coffee/extendView/commonPage.coffee [hunk 2]
```diff
@@ -4,21 +4,33 @@ extendView['commonPage'] = {
...
     $('.show-areas', @$el).show()
     $('.area-toolbar', @$el).removeClass('shown')
 
-  addConfigurationButton: () ->
-    pageConfigurationButtonViewClass = appConfigurationView.getConfiguration(@options.entityType, 'addConfigurationButton')
+  ###
+   * Add layout button (edit and delete)
+  ###
+  addLayoutButton: () ->
+    pageConfigurationButtonViewClass = appConfigurationView.getConfiguration(@options.entityType, 'addLayoutButton')
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/extendView/commonPage.coffee]: can you rename this variable

File: BackofficeBundle/Resources/public/coffee/page/NodeView.coffee [hunk 3]
```diff
@@ -39,7 +42,8 @@ NodeView = OrchestraView.extend(
         element: @options.node
       )
       @addAreasToView(@options.node.get('areas'))
-      @addConfigurationButton()
+      if @options.configuration.get('links')._self_form?
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/page/NodeView.coffee]: you should also test _self_delete

File: BackofficeBundle/Resources/public/coffee/viewConfigurator.coffee [hunk 4]
```diff
@@ -10,7 +10,7 @@ OrchestraViewConfigurator = ->
     'showAreaFlexToolbar': OpenOrchestra.AreaFlex.AreaFlexToolbarView
     'addBlock': BlockView
     'addButtonAction': TableviewAction
-    'addConfigurationButton': PageConfigurationButtonView
+    'addLayoutButton': OpenOrchestra.Page.PageLayoutButtonView
```
Comment (itkg-nanne) [inline BackofficeBundle/Resources/public/coffee/viewConfigurator.coffee]: rename by addPageLayoutButton

File: BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+###*
+ * @class PageLayoutButtonView
+###
+class OpenOrchestra.Page.PageLayoutButtonView extends OrchestraView
+
+  events:
+    'click span': 'configurationPage'
+    'click .ajax-delete': 'deleteButton'
+
+  initialize: (options) ->
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee]: js doc

File: BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      deleteUrl: @options.deleteUrl
+      confirmText: @options.confirmText
+      confirmTitle: @options.confirmTitle
+      redirectUrl: @options.redirectUrl
+    )
+    @$el.attr('data-widget-index', @options.widget_index)
+    OpenOrchestra.RibbonButton.ribbonFormButtonView.setFocusedView(@, '.ribbon-form-button')
+    return
+
+  configurationPage: () ->
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee]: js doc

**Comment by KevinArtus** [comment]
@itkg-nanne @alavieille I corrected everything you asked. Could you check it please ?

File: BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+   * Render area
+  ###
+  render: ->
+    @setElement @renderTemplate('OpenOrchestraBackofficeBundle:BackOffice:Underscore/widgetPageLayoutButton',
+      deleteUrl: @options.deleteUrl
+      confirmText: @options.confirmText
+      confirmTitle: @options.confirmTitle
+      redirectUrl: @options.redirectUrl
+    )
+    @$el.attr('data-widget-index', @options.widget_index)
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee]: this parameter is not used

File: BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  events:
+    'click span': 'configurationPage'
+    'click .ajax-delete': 'deleteButton'
+
+  initialize: (options) ->
+    @options = @reduceOption(options, [
+      'configuration'
+      'viewContainer'
+      'entityType'
+      'widget_index'
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee]: remove this

File: BackofficeBundle/Resources/public/coffee/extendView/commonPage.coffee [hunk 9]
```diff
@@ -4,21 +4,33 @@ extendView['commonPage'] = {
...
-    pageConfigurationButtonViewClass = appConfigurationView.getConfiguration(@options.entityType, 'addConfigurationButton')
-    new pageConfigurationButtonViewClass(@addOption(
+  ###
+   * Add layout button (edit and delete)
+  ###
+  addPageLayoutButton: () ->
+    pageLayoutButtonViewClass = appConfigurationView.getConfiguration(@options.entityType, 'addPageLayoutButton')
+    new pageLayoutButtonViewClass(@addOption(
       viewContainer: @
       widget_index: 2
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/extendView/commonPage.coffee]: remove this

File: BackofficeBundle/Resources/public/coffee/page/NodeView.coffee [hunk 10]
```diff
@@ -39,7 +42,8 @@ NodeView = OrchestraView.extend(
         element: @options.node
       )
       @addAreasToView(@options.node.get('areas'))
-      @addConfigurationButton()
+      if @options.configuration.get('links')._self_form? and @options.configuration.get('links')._self_delete?
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/page/NodeView.coffee]: if you can add only the button delete and not edit, you couldn't do this

File: BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+      'confirmTitle'
+      'redirectUrl'
+    ])
+    @loadTemplates [
+      "OpenOrchestraBackofficeBundle:BackOffice:Underscore/widgetPageLayoutButton"
+    ]
+    return
+
+  ###
+   * Render area
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/page/PageLayoutButtonView.coffee]: area ??

File: BackofficeBundle/Resources/public/coffee/AdminFormView.coffee [hunk 12]
```diff
@@ -35,8 +31,6 @@ AdminFormView = OrchestraView.extend(
         originalButton = $('.submit_form', response)
         button = originalButton.clone().attr('data-clone', originalButton.attr('id')).removeAttr('id')
         actionButtons = $('<div>')
-        if viewContext.deleteButton
-          actionButtons.append(viewContext.renderTemplate('OpenOrchestraBackofficeBundle:BackOffice:Underscore/deleteButton', viewContext.options))
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/AdminFormView.coffee]: remove also deleteButton file

File: BackofficeBundle/Resources/public/coffee/AdminFormView.coffee [hunk 13]
```diff
@@ -1,17 +1,13 @@
...
       'title'
       'url'
       'extendView'
       'method'
       'entityType'
     ])
-    @deleteButton = @options.deleteUrl && @options.confirmText && @options.confirmTitle
     @method = if @options.method then @options.method else 'GET'
     @events = @events || {}
     @loadTemplates [
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/AdminFormView.coffee]: remove loadTemplates deleteButton

**Comment by KevinArtus** [comment]
@alavieille I changed test to display buttons.

**Comment by KevinArtus** [comment]
@alavieille I removed @loadTemplates methods

**Comment by KevinArtus** [comment]
@alavieille @itkg-nanne Js doc added

File: BackofficeBundle/Resources/public/coffee/page/NodeView.coffee [hunk 14]
```diff
@@ -29,6 +31,7 @@ NodeView = OrchestraView.extend(
     return
 
   render: ->
+    viewContext = @
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/coffee/page/NodeView.coffee]: you do not use this variable

**Comment by KevinArtus** [comment]
@alavieille I removed unused variable, can you validate please?

**Comment by alavieille** [comment]
can you add OO-BCBREAK for view than you have deleted adter :+1:

**Pull request merged**